Library/Homebrew/keg.rb

```diff
@@ -157,6 +158,25 @@ def self.find_some_installed_dependents(kegs)
     nil
   end
 
+  # Finds kegs that were installed as dependencies of other kegs,
+  # but that are no longer required by any explicitly-installed kegs.
+  def self.orphaned
+    all.select(&:orphaned?)
+  end
+
+  def orphaned?
+    return false if might_have_been? :installed_on_request
+    installed_dependents.none? { |d| d.might_have_been? :installed_on_request }
+  end
+
+  def might_have_been?(key)
```

Comment:
Yeh, I think so, thanks. I'd be tempted to just call the tab methods, though; I think that's going to be more readable.